modules/developers-guide/examples/graphql/DML_data.graphql

```diff
@@ -1,72 +1,312 @@
+        item2
+      }
+     }
+   }
 }
 
+# update one book, adding an ISBN value
+# *** USED IN QUICKSTART ***
 mutation updateOneBook {
-  moby: updateBooks(value: {title:"Moby Dick", author:"Herman Melville", isbn: "9780140861723"}, ifExists: true ) {
+  moby: updatebook(value: {title:"Moby Dick", author:"Herman Melville", isbn: "9780140861723"}, ifExists: true ) {
     value {
       title
       author
       isbn
     }
   }
+}
 
-mutation insertPap {
-  pap: insertBooks(value: {title:"Pride and Prejudice", author:"Jane Austen"}) {
-    value {
-      title
+# update one book, adding a SET (genre)
+# *** ADVANCED ***
+mutation updateOneBookAgain {
+ moby: updatebook(value: {title:"Moby Dick", author:"Herman Melville", genre: ["Drama", "Classic lit"]}, ifExists: true ) {
+   value {
+     title
+     author
+     genre
+   }
+ }
+}
+
+# get one book using the primary key title with a value
+# *** USED IN QUICKSTART ***
+query oneBook {
+    book (value: {title:"Moby Dick"}) {
+      values {
+      	title
+      	author
+      }
+    }
+}
+
+# get 3 books using the primary keys with an "in" filter clause of the primary key title
+# *** ADVANCED ***
+query readThreeBooks {
+  book(filter: { title: { in: ["Native Son", "Moby Dick", "Catch-22"] } } ) {
+      values {
+      	title
+ 			  author
+     }
+   }
+}
+
+# query the author to see the UDT
+# *** ADVANCED ***
+query getReaderWithUDT{
+  reader(value: { name:"Allen Ginsberg" user_id: "e0ed81c3-0826-473e-be05-7de4b4592f64" }) {
+    values {
+      name
+      birthdate
+      addresses {
+        street
+        city
+        zip
+      }
     }
   }
 }
 
-mutation deleteOneBook {
-  PaP: deleteBooks(value: {title:"Pride and Prejudice"}, ifExists: true ) {
+# query a badge record that has a MAP (earned) with only the partition key
+# *** ADVANCED ***
+query oneGoldBadge {
+  badge(value: { badge_type: "Gold" } ) {
+      values {
+      	badge_type
+        badge_id
+        earned {
+        key
+        value
+      }
+     }
+  }
+}
+
+# // NEED EXAMPLES FOR USING THE FILTERS: EQ NEQ,GT, GTE,LT, LTE, IN, CONTAINS (FOR MAP?)
+
+# query a badge record that has a MAP (earned) with the partition key and the clustering key
+# *** ADVANCED ***
+query oneGold100Badge {
+  badge(filter: { badge_type: {eq:"Gold"} badge_id: {eq:100}} ) {
+      values {
+      	badge_type
+        badge_id
+      	earned {
+          key
+          value
+        }
+     }
+   }
+}
+# query a badge record that has a MAP (earned) with the partition key and the clustering key
+# gt:100 will fail
+# *** ADVANCED ***
+query oneGold100BadgeGTFail {
+  badge(filter: { badge_type: {eq:"Gold"} badge_id: {gt:100}} ) {
+      values {
+      	badge_type
+        badge_id
+      	earned {
+          key
+          value
+        }
+     }
+   }
+}
+
+# query a badge record that has a MAP (earned) with the partition key, clustering key, and a MAP key
+# CREATE INDEX badge_idx ON library.badge(KEYS(earned));
```

Comment:
Is the need for an index explained somewhere?

Reply:
Also I'm not familiar, will these comments show up?

Reply:
I do explain the indexes in the text. And this file is basically my import/export file for graphql - it isn't in the docs.

Reply:
Got it